### PR TITLE
Remove components-invalidation-impl-fcm_network_handler.cc.patch

### PR DESCRIPTION
**Pull request by Gyuyoung**
The patch file can be replace by overriding StartListening() in
//brave/chromium_src.

Resolves brave/brave-browser#12685.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: chromium_src/components/invalidation/impl/fcm_network_handler.cc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include "../../../../../components/invalidation/impl/fcm_network_handler.cc"
+#undef StartListening
+
+namespace syncer {
+
+void FCMNetworkHandler::StartListening() {
+  return;
+}
+
+}  // namespace syncer
```
Comment (mkarolin) [inline chromium_src/components/invalidation/impl/fcm_network_handler.cc:18]: Lint: new line after.

File: chromium_src/components/invalidation/impl/fcm_network_handler.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_INVALIDATION_FCM_NETWORK_HANDLER_H_
+
+#include "components/invalidation/impl/fcm_sync_network_channel.h"
+
+#define StartListening StartListening_ChromiumImpl(); \
+  virtual void StartListening
+#include "../../../../../components/invalidation/impl/fcm_network_handler.h"
+#undef StartListening
+
+#endif  // BRAVE_CHROMIUM_SRC_COMPONENTS_INVALIDATION_FCM_NETWORK_HANDLER_H_
```
Comment (mkarolin) [inline chromium_src/components/invalidation/impl/fcm_network_handler.h:16]: Lint: new line after.

File: chromium_src/components/invalidation/impl/fcm_network_handler.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_INVALIDATION_FCM_NETWORK_HANDLER_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_INVALIDATION_FCM_NETWORK_HANDLER_H_
+
+#include "components/invalidation/impl/fcm_sync_network_channel.h"
+
+#define StartListening StartListening_ChromiumImpl(); \
+  virtual void StartListening
```
Comment (mkarolin) [inline chromium_src/components/invalidation/impl/fcm_network_handler.h:12]: Doesn't need `virtual` - it already has `override` in the original file.

File: chromium_src/components/invalidation/impl/fcm_network_handler.cc (hunk 1 above)
Comment (Gyuyoung) [reply chromium_src/components/invalidation/impl/fcm_network_handler.cc:18]: Done.

File: chromium_src/components/invalidation/impl/fcm_network_handler.h (hunk 3 above)
Comment (Gyuyoung) [reply chromium_src/components/invalidation/impl/fcm_network_handler.h:12]: Ah, done.

File: chromium_src/components/invalidation/impl/fcm_network_handler.h (hunk 2 above)
Comment (Gyuyoung) [reply chromium_src/components/invalidation/impl/fcm_network_handler.h:16]: done

**Review by mkarolin — APPROVED** [review]
LGTM

**Comment by mkarolin** [comment]
Just to follow up: @jumde the FCM disablement unit test was removed here https://github.com/brave/brave-core/pull/3390/files?file-filters%5B%5D=.cc#diff-f6da31f7767584ee44a09726c06bd93a5613415f1bd69d49430b94bfd7861e95L235. Do we even need this change any more? If we do, maybe we should bring back the unit test.

**Comment by Gyuyoung** [comment]
@mkarolin Thanks for reviewing this PR. Merging...

**Pull request merged**